compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -706,17 +706,7 @@ impl<'tcx> Inliner<'tcx> {
         });
 
         // Copy only unevaluated constants from the callee_body into the caller_body.
-        // Although we are only pushing `ConstKind::Unevaluated` consts to
-        // `required_consts`, here we may not only have `ConstKind::Unevaluated`
-        // because we are calling `instantiate_and_normalize_erasing_regions`.
-        caller_body.required_consts.extend(callee_body.required_consts.iter().copied().filter(
-            |&ct| match ct.const_ {
-                Const::Ty(_) => {
-                    bug!("should never encounter ty::UnevaluatedConst in `required_consts`")
```

Comment:
I don't know why this was made to ICE, but if we push `ty::ConstKind::Param` to `required_consts` (as I think we have to), then of course they can reach this position here, so I had to remove this `bug!`.

Reply:
we create `callee_body` via `try_instantiate_mir_and_normalize_erasing_regions`, which evaluated all constants. I guess we forgot to make it evaluate `required_consts`?

Reply:
> as I think we have to

uh, why do we need to have them in there? shouldn't these be evaluated at the caller site? or is the issue that even those constants may be generic?

Reply:
> a parameter can become a concrete constant with an error later, right?

yes. But in that case an error has happened higher up in the call stack when evaluating the constant. We should have an affected example before adding it here imo. Otherwise we're just adding all params.

Reply:
> in that case an error has happened higher up in the call stack when evaluating the constant

The question is, does that reliably prevent the code further down the call stack from being evaluated?

But I'm also okay with only adding `Error` for now to fix the immediate ICE.

Reply:
I hope so.

Adding error is fine.